sktime/utils/validation/forecasting.py

```diff
@@ -101,16 +106,28 @@ def check_X(
     )
 
 
-def check_y(y, allow_empty=False, allow_constant=True, enforce_index_type=None):
```

Comment:
This would seem to make sense, but is actually not what we want, the reason is that `y` is here defined as the expected input type for forecasters, which we always want to be a pd.Series and univariate.

Reply:
I think my tweaks to the function definition of `check_y` preserve the original functionality. I just added two new keyword arguments (allow_numpy and enforce_univariate). I set the defaults to be the original values used in the call to `check_series` so that any calls to the function that don't reference the new arguments should work as they did before (not allow NumPy arrays and enforce univariate structure). Adding the parameters makes it easier to re-use the rest of the functionality in the performance metrics (and potentially other places) where we want to allow multivariate Y data or NumPy arrays (or both). 

Does that make sense?

Reply:
I see. The issue is that `check_y` here defines the allowed input formats for `y` in all forecasters (it defines the [API contract](https://en.wikipedia.org/wiki/Design_by_contract)). As we currently specify the API, `y` should always be a univariate `pd.Series` and never a numpy array or multivariate. 

Since we follow scikit-learn's API for the scoring functions, I suggest we simply reuse their input checks (e.g. [_check_reg_targets](https://github.com/scikit-learn/scikit-learn/blob/95119c13af77c76e150b753485c662b7c52a41a2/sklearn/metrics/_regression.py#L335) or a copy of that since it's part of their private functionality).

Reply:
@mloning I'm fine with that approach, but what about leaving the `enforce_univariate` parameter argument to `check_y` (I'm removing the numpy array argument either way). Adding multivariate forecasting down the road would be interesting so leaving the `enforce_univariate` option retains the functionality that would be needed down the road.

Reply:
I may make more sense to add a `check_Y` function for multivariate forecasting in that case.